src/lib/proof-system/zkprogram.ts

```diff
@@ -1487,20 +1533,37 @@ function getMaxProofsVerified(methodIntfs: MethodInterface[]) {
   ) as any as 0 | 1 | 2;
 }
 
-function fromFieldVars<T>(type: ProvablePure<T>, fields: MlFieldArray) {
-  return type.fromFields(MlFieldArray.from(fields));
+function fromFieldVars<T>(type: Provable<T>, fields: MlFieldArray) {
+  return type.fromFields(MlFieldArray.from(fields), []);
 }
+
 function toFieldVars<T>(type: ProvablePure<T>, value: T) {
   return MlFieldArray.to(type.toFields(value));
 }
 
-function fromFieldConsts<T>(type: ProvablePure<T>, fields: MlFieldConstArray) {
+function fromFieldConsts<T>(type: any, fields: MlFieldConstArray) {
   return type.fromFields(MlFieldConstArray.from(fields));
 }
-function toFieldConsts<T>(type: ProvablePure<T>, value: T) {
+
+function fromFieldAndAuxConsts<T>(
+  type: Provable<T>,
+  fields: MlFieldConstArray,
+  aux: any[] = []
```

Comment:
nit: we could do `unknown` too, maybe a bit more fitting

Reply:
The reason I made the aux type `any[]` was to match the return value from `toAuxiliary`.
```ts
(property) toAuxiliary: (value?: T | undefined) => any[]
A function that takes value (optional), an element of type T, as argument and returns an array of any type that make up the "auxiliary" (non-provable) data of value.

@param value
the element of type T to generate the auxiliary data array from, optional. If not provided, a default value for auxiliary data is returned.

@return — An array of any type describing how this T element is made up of "auxiliary" (non-provable) data.
```

Reply:
That's right! Should be okay then